### PR TITLE
Add AssertEqualsToSameRector

### PR DESCRIPTION
**Pull request by dereuromark**
Closes https://github.com/rectorphp/rector/issues/3742

EDIT by @TomasVotruba - added Github [closing keyword](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords) so issue is closed by Github on merge

### CODE REVIEW

**Comment by peter279k** [comment]
Coding style checking is failed and it should be fixed.

**Comment by peter279k** [comment]
Here is the coding style checking report:

```
1) rules/phpunit/src/Rector/SpecificMethod/AssertEqualsToSameRector.php

    ---------- begin diff ----------
--- Original
+++ New
@@ -1,4 +1,6 @@
-<?php declare(strict_types=1);
+<?php
+
+declare(strict_types=1);

 namespace Rector\PHPUnit\Rector\SpecificMethod;
    ----------- end diff -----------


Applied checkers:

 * PhpCsFixer\Fixer\PhpTag\BlankLineAfterOpeningTagFixer
```

**Comment by peter279k** [comment]
Rector CI is failed and please look at this:

https://github.com/rectorphp/rector/pull/3958/checks?check_run_id=984627715

**Comment by peter279k** [comment]
The static code analysis checking with PHPStan has been failed and please look at this:

https://github.com/rectorphp/rector/pull/3958/checks?check_run_id=984627927

File: rules/phpunit/tests/Rector/AssertEqualsToSameRector/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        $expectedFloat = 1.1;
+        $this->assertSame($expectedFloat, $float);
+
+        $string = 'abc';
+        $expectedString = 'abc';
+        $this->assertSame($expectedString, $string);
+    }
+}
+
+?>
```
Comment (TomasVotruba) [inline rules/phpunit/tests/Rector/AssertEqualsToSameRector/Fixture/fixture.php.inc:51]: How can be sure from tests that the null/bool cases are skipped?

**Comment by TomasVotruba** [comment]
@dereuromark Thank you for your work, I like the simplicity :+1:

**Comment by TomasVotruba** [comment]
@peter279k Thank you for review and feedback

File: rules/phpunit/tests/Rector/AssertEqualsToSameRector/Fixture/fixture.php.inc (hunk 1 above)
Comment (dereuromark) [reply rules/phpunit/tests/Rector/AssertEqualsToSameRector/Fixture/fixture.php.inc:51]: I found it quite clear from the specific type map.
But sure, I can add a skip one too.

**Pull request merged**